File: backend/internal/dns/record.go
package dns

import "time"

type Record struct {
	Dns  DnsRecord      `json:"dnsRecord"`
	Meta RecordMetadata `json:"metadata"`
}

type DnsRecord struct {
	Name  string `json:"name"`
	Type  string `json:"type"`  // A or CNAME
	Value string `json:"value"` // IP or hostname
}

type RecordMetadata struct {
	ContainerID   string    `json:"containerId"`
	ContainerName string    `json:"containerName"`
	Created       time.Time `json:"created"`
	Hostname      string    `json:"hostname"`
	Force         bool      `json:"force"`
}

File: frontend/src/utils/record.ts
import { RecordEntry } from '../types';

export const getRecordKey = (record: RecordEntry) => `${record.dnsRecord.name}|${record.dnsRecord.type}|${record.dnsRecord.value}|${record.metadata.containerId}`;

export function enrichSearchable(records: RecordEntry[]) {
  return records.map((r) => {
    const { dnsRecord, metadata } = r;
    const searchText = [
      dnsRecord.name,
      dnsRecord.type,
      dnsRecord.value,
      metadata.containerId,
      metadata.containerName,
      metadata.hostname,
      metadata.created,
      metadata.force?.toString(),
    ]
      .filter(Boolean)
      .join(' ')
      .toLowerCase();

    return {
      ...r,
      searchable: searchText,
    };
  });
}
